File: POLIMORFISMO/Ejercicio1/Videojuego.py
from multimethod import multimethod
class Videojuego:
    def __init__(self, nombre="MarioKart", plataforma="Nintendo", cantidadJugadores=4):
        self.__nombre = nombre
        self.__plataforma = plataforma
        self.__cantidadJugadores = cantidadJugadores
    @multimethod
    def agregarJugador(self,uno=1):
        self.__cantidadJugadores += uno
        return self.__cantidadJugadores
    @multimethod
    def agregarJugador(self, otro):
        self.__cantidadJugadores += otro
        return self.__cantidadJugadores

    def mostrar(self):
        print(f"Nombre: {self.__nombre} --- Plataforma: {self.__plataforma} --- Jugadores: {self.__cantidadJugadores}")
V1 = Videojuego()
V1.mostrar()
print(V1.agregarJugador(1))
V2 = Videojuego("Call of Duty", "PC", 2)
V2.mostrar()
print(V2.agregarJugador(5))

File: POLIMORFISMO/Ejercicio1/Videojuego.java
public class Videojuego {
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;
    //metodo constructor 

    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }

    public Videojuego(String nombre, int cantidadJugadores) {
        this.nombre = nombre;
        this.cantidadJugadores = cantidadJugadores;
        this.plataforma="desconocido";
    }

    public Videojuego(String plataforma) {
        this.plataforma = plataforma;
        this.nombre="desconocido";
        this.cantidadJugadores=1; 
    }
    
    public String mostrar(){
        return String.format("Nombre:%s ,Plataforma:%s ,Cantidad de jugadores:%d",this.nombre,this.plataforma,this.cantidadJugadores);
    }
    public int agregarJugadores(){
        return this.cantidadJugadores+=1;
    }
    public int agregarJugadores(int jugadores){
        return this.cantidadJugadores+=jugadores;
    }
}

